src/System.Management.Automation/FormatAndOutput/common/ColumnWidthManager.cs

```diff
@@ -107,18 +107,31 @@ private bool AssignColumnWidths(Span<int> columnWidths)
             }
 
             // we still have room and we want to add more width
+            int unfixedColumnCount = 0;
+            foreach (bool columnIsFixed in fixedColumn)
+            {
+                if (!columnIsFixed)
+                {
+                    unfixedColumnCount++;
+                }
+            }
 
-            while (availableWidth > 0)
+            if (unfixedColumnCount > 0)
```

Comment:
I'd suggest to count in cycle above and also consider using stackalloc:
```c#
            // we have columns with no width assigned
            // remember the columns we are trying to size
            // assign them the minimum column size
            int unfixedColumnCount = 0;
            ReadOnlySpan<bool> fixedColumn = stackalloc bool[columnWidths.Length];
            for (int k = 0; k < columnWidths.Length; k++)
            {
                if (columnWidths[k] == 0)
                {
                    columnWidths[k] = _minimumColumnWidth;
                    unfixedColumnCount++;
                }
                else
                {
                    fixedColumn[k] = true;
                }
            }
```

Reply:
I took your suggestion with one difference: instead of a `ReadOnlySpan` I used a `Span` since the span is being modified after it.